dir.c

```diff
@@ -1653,11 +1673,11 @@ to_be_ignored(int e)
 }
 
 #ifdef _WIN32
-#define STAT(p, s)	rb_w32_ustati128((p), (s))
-#undef lstat
-#define lstat(p, s)	rb_w32_ulstati128((p), (s))
+#define STAT(args)	(int)(VALUE)nogvl_stat((void *)&(args))
+#define LSTAT(args)	(int)(VALUE)nogvl_lstat((void *)&(args))
 #else
-#define STAT(p, s)	stat((p), (s))
+#define STAT(args)	IO_WITHOUT_GVL_INT(nogvl_stat, (void *)&(args))
+#define LSTAT(args)	IO_WITHOUT_GVL_INT(nogvl_lstat, (void *)&(args))
```

Comment:
These `(void *)` casts are needed?

Reply:
After some research, I do not think so.  Implicit casts to and from `void *` are allowed, unless they are part of a function pointer.  I found other unnecessary casts to and from `void *` in `dir.c`, and tried to fix those as well (let's see if CI likes it).